docs/csharp/misc/cs0501.md

```diff
@@ -13,15 +13,42 @@ ms.assetid: 3ff45208-5b9b-42f6-8a12-1eb38a665f33
 'member function' must declare a body because it is not marked abstract, extern, or partial  
   
  Nonabstract methods must have implementations.  
+
+## Explanation
+
+In C#, methods/functions that are a part of a class must have a "body", or implementation. The compiler needs to know what should happen when these methods are called, so that it knows what to execute. A method with on body is not acceptable to the compiler because it wants to avoid confusion about the intent of the code.
```

Comment:
nit, typo:

```suggestion
In C#, methods/functions that are a part of a class must have a "body", or implementation. The compiler needs to know what should happen when these methods are called, so that it knows what to execute. A method with no body is not acceptable to the compiler because it wants to avoid confusion about the intent of the code.
```